### DISCUSSION TITLE
How to create One Note Pages

### DISCUSSION THREAD

**Discussion by carogomezt** [microsoftgraph/msgraph-sdk-python#955] [Q&A]
Hi everyone!
I was trying to create a OneNote Page using this code:

```python
title = "Test Title"
html_content = """
    <!DOCTYPE html>
    <html>
    <head>
        <title>Sample Page</title>
    </head>
    <body>
        <p>This is a sample page created using the Microsoft Graph API.</p>
    </body>
    </html>
    """

multipart_data = (
    "--boundary_string\r\n"
    'Content-Disposition: form-data; name="Presentation"\r\n'
    "Content-Type: text/html\r\n\r\n"
    f"{html_content}\r\n"
    "--boundary_string--"
).encode("utf-8")
content = bytes(multipart_data)

request_body = OnenotePage(title=title, content=content)
response = (
    await graph_client.sites.by_site_id(site_id)
    .onenote.sections.by_onenote_section_id(section_id)
    .pages.post(request_body)
)

```

But I'm getting the following error message:

`message='Page create requests require the content to be multipart, with a presentation part.`

Do you know what I might be missing? or do you have an example of how to create a OneNote page using the Python SDK? I have been trying to find one in the documentation, but I haven't been successful.

**Comment by meaniebeanie22** [comment]
The best I can find is this reference to the javascript version on stackoverflow - I think a similar idea would work but I'm unsure of how to implement it.
https://stackoverflow.com/questions/51145245/multipart-http-request-with-microsoft-graph-javascript-sdk
I'm encountering the same issue - would someone with a clue be able to help? :)
